### PR TITLE
small changes to allow connections to memcached through unix socket

### PR DESCRIPTION
**Pull request by stratsimir**
Some memcached setups allow only connections through unix domain sockets. Currently PrestaShop doesn't allow to set unix socket path for host and "0" for port. This little patch fixes it.

### CODE REVIEW

**Comment by xBorderie** [comment]
Hey @stratsimir , thank you for this PR!

Could you follow the [guidelines for PR and commit titles](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)? 

Thank you!

File: controllers/admin/AdminPerformanceController.php [hunk 1]
```diff
@@ -1029,12 +1029,12 @@ public function displayAjaxTestServer()
...
             $type = Tools::getValue('type', '');
 
-            if ($host != '' && $port != 0) {
+            if ($host != '') {
                 $res = 0;
 
                 if ($type == 'memcached') {
                     if (extension_loaded('memcached') &&
-                        @fsockopen($host, $port)
+                        (strpos($host, '/') === 0 || @fsockopen($host, $port)) //strpos to check whether the value is path (unix socket)
```
Comment (jocel1) [inline controllers/admin/AdminPerformanceController.php]: @fsockopen($host, (($port === 0)?-1:$port)) should work without the strpos hack

File: controllers/admin/AdminPerformanceController.php [hunk 2]
```diff
@@ -1044,7 +1044,7 @@ public function displayAjaxTestServer()
                 } else {
                     if (function_exists('memcache_get_server_status') &&
                         function_exists('memcache_connect') &&
-                        @fsockopen($host, $port)
+                        (strpos($host, 'unix://') === 0 || @fsockopen($host, $port))
```
Comment (jocel1) [inline controllers/admin/AdminPerformanceController.php]: same here

**Comment by julienbourdeau** [comment]
Hi,

Can you update your PR so I can merge it right away?

**Comment by ingmarheinrich** [comment]
What about this small change?

**Comment by xBorderie** [comment]
@ingmarheinrich We'd need @stratsimir to update his modifications in order to merge it.
Does it solve an issue of yours?

**Comment by thierrymarianne** [comment]
Closing as it breaks the build.

**Comment by neoteknic** [comment]
Please reopen it, it's useless to use the TCP stack in local (TCP overload). Add unix socket support.

**Pull request closed without merging**